NextcloudTalk/UserStatusOptionsSwiftUI.swift

```diff
@@ -0,0 +1,128 @@
+//
+// Copyright (c) 2023 Lukas Lauerer <lukas.lauerer@gmx.net>
+//
+// Author Lukas Lauerer <lukas.lauerer@gmx.net>
+//
+// GNU GPL version 3 or any later version
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+//
+
+import SwiftUI
+import NextcloudKit
+
+struct UserStatusOptionsSwiftUI: View {
+    @Environment(\.dismiss) var dismiss
+
+    @Binding var changed: Bool
+    @Binding var userStatus: NCUserStatus
+
+    @State private var options: [DetailedOption] = []
+    @State private var isLoading: Bool = true
+
+    init(changed: Binding<Bool>, userStatus: Binding<NCUserStatus>) {
+        _changed = changed
+        _userStatus = userStatus
+    }
+
+    var body: some View {
+            VStack {
+                List(options, id: \.self) { option in
+                    Button(action: {
+                        setActiveUserStatus(userStatus: option.identifier)
+                    }) {
+                        HStack(spacing: 15) {
+                            Image(uiImage: option.image)
+                            VStack(alignment: .leading) {
+                                Text(option.title)
+                                    .font(.system(size: 18))
+                                    .foregroundColor(.primary)
+                                if option.subtitle != nil && !option.subtitle.isEmpty {
```

Comment:
```suggestion
                                if !option.subtitle?.isEmpty {
```

Not tested, but shouldn't something like this work?

Reply:
Unfortunately at least not as written above. for the if statement it cant be optional.